daemon/oci_linux.go

```diff
@@ -76,10 +78,30 @@ func setResources(s *specs.Spec, r containertypes.Resources) error {
 		specResources.Devices = s.Linux.Resources.Devices
 	}
 
+	if len(r.GPUs) > 0 {
+		if !experimental {
```

Comment:
Would it make sense to push this experimental check up the stack so we don't have to deal with it here?
Also, do you really feel this should be experimental? Are you worried the API is not sufficient?

Reply:
> Would it make sense to push this experimental check up the stack so we don't have to deal with it here?

I initially considered what you said previously, that is to say to create a function `setResourcesExperimental` and put it here after `setResources`:
https://github.com/moby/moby/blob/658675e7d6fff1e4ef7c153398c25fbbe52a5080/daemon/oci_linux.go#L737-L739
Then I had two options:
```golang
if daemon.HasExperimental() {
        if err := setResourcesExperimental(...) {
                return nil, fmt.Errorf(...)
        }
}
```
That's not great since we lose the ability to warn/error if GPUs are requested but the experimental flag is not set.

The other option would have been:
```
if err := setResourcesExperimental(...,  daemon.HasExperimental()) {
        return nil, fmt.Errorf(...)
}
```
But it's bizarre to have a function `setResourcesExperimental` and pass a boolean flag for this function to really do something.

Let me know if you see another option.